### PR TITLE
Remove unused methods

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/IO/PersistedFiles.Unix.cs [hunk 1]
```diff
@@ -10,75 +10,6 @@ namespace System.IO
...
     {
-        private static string? s_userProductDirectory;
-
-        /// <summary>
-        /// Get the location of where to persist information for a particular aspect of the framework,
-        /// such as "cryptography".
-        /// </summary>
-        /// <param name="featureName">The directory name for the feature</param>
-        /// <returns>A path within the user's home directory for persisting data for the feature</returns>
-        internal static string GetUserFeatureDirectory(string featureName)
```
Comment (MichalStrehovsky) [inline src/System.Private.CoreLib/shared/System/IO/PersistedFiles.Unix.cs]: Doesn't System.Security.Cryptography.X509Certificates in CoreFX use this?

Comment (marek-safar) [reply src/System.Private.CoreLib/shared/System/IO/PersistedFiles.Unix.cs]: Could we move the corefx part to corefx as partial class of this type to make it more obvious?

Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/IO/PersistedFiles.Unix.cs]: corefx uses everything in this file.

Comment (marek-safar) [reply src/System.Private.CoreLib/shared/System/IO/PersistedFiles.Unix.cs]: undone the change

**Pull request merged**